blender/arm/material/make_mesh.py

```diff
@@ -708,6 +707,21 @@ def make_forward_base(con_mesh, parse_opacity=False, transluc_pass=False):
             else:
                 frag.add_uniform('vec2 lightProj', link='_lightPlaneProj', included=True)
                 frag.add_uniform('samplerCubeShadow shadowMapPoint[1]', included=True)
+        if '_MicroShadowing' in wrd.world_defs:
+            frag.add_include('std/gbuffer.glsl')
+            frag.add_uniform('sampler2D gbuffer1')
+            frag.add_uniform('sampler2D gbufferD')
+            frag.add_uniform('vec3 eyeLook')
+            frag.add_uniform('vec2 cameraProj')
+            frag.add_in('vec2 texCoord')
+            frag.add_in('vec3 viewRay')
+            frag.write('vec4 g1 = textureLod(gbuffer1, texCoord, 0.0);')
+            frag.write('vec2 occspec = unpackFloat2(g1.a);')
+            frag.write('float depth = textureLod(gbufferD, texCoord, 0.0).r * 2.0 - 1.0;')
```

Comment:
What are all these calculations for if you only use  `occspec` later?

Reply:
Seems like the GitHub previews is bit misleading, I meant `float depth`, `p`, `v` and everything above that is only required for those calculations.